synapse/tchap/event_rules.py

```diff
@@ -27,6 +28,9 @@
 class TchapEventRules(object):
     def __init__(self, config):
         self.id_server = config["id_server"]
+        self.domains_forbidden_when_restricted = config.get(
+            "domains_forbidden_when_restricted", [],
+        )
```

Comment:
Not sure the requirement specifies either, I went for a blacklist because the `DomainRuleChecker` uses a blacklist (as far as I've seen) which would make setting the config easier.